core/src/main/java/com/segment/analytics/kotlin/core/platform/plugins/SegmentDestination.kt

```diff
@@ -99,4 +103,20 @@ class SegmentDestination : DestinationPlugin() {
     override fun flush() {
         pipeline.flush()
     }
+
+    internal fun <T: BaseEvent> configureCloudDestination(event: T): T {
+        val enabledDestinations = analytics.findAll(DestinationPlugin::class).filter { it.enabled }
```

Comment:
doing a `findAll` for each of the enqueuing event could be a bottle neck, though the total number of plugins could be small, it does not scale well.
is it possible to make `SegmentDestination` a `Subscriber` of `System` state? we cache the `enabledDestinations`, and update it when there is plugin changes. 
just a suggestion, may not be feasible 😄

Reply:
lemme try that

Reply:
i went with a different route here so as to make it similar to the swift code